src/main/java/use_cases/session_load/SessionLoadUCI.java

```diff
@@ -0,0 +1,48 @@
+package use_cases.session_load;
+
+import entities.SessionStorage;
+import views.file_session_storage.SessionStorageG;
+
+import java.io.IOException;
+
+public class SessionLoadUCI implements SessionLoadIB {
+    private final SessionStorageG storageGateway;
+    private final SessionLoadOB presenter;
+    private final SessionStorage sessionToModify;
+
+
+    public SessionLoadUCI(SessionStorageG storageGateway, SessionLoadOB presenter, SessionStorage sessionToModify) {
+        this.storageGateway = storageGateway;
+        this.presenter = presenter;
+        this.sessionToModify = sessionToModify;
+    }
+
+    /**
+     * Either loads a SessionStorage from a file, or creates a new empty one.
+     * Then, mutates this.sessionToModify to reflect the data loaded/created.
+     *
+     * @param inputData the user input data
+     * @return a SessionLoadOD object with the output data to be used by the View
+     * @throws SessionLoadException if the user chose to load a file and there is an error while opening it
+     */
+    @Override
+    public SessionLoadOD load(SessionLoadID inputData) throws SessionLoadException {
+        if (inputData.getIsNew()) {
+            // Create a new empty SessionStorage
+            SessionStorage new_session = new SessionStorage();
+            this.sessionToModify.copyDataFrom(new_session);
+            return this.presenter.displaySuccess("Created new session successfully", this.sessionToModify);
+        } else {
+            // Load a SessionStorage from a file
+            try {
```

Comment:
Why did you decide to only handle some errors and throw SessionLoadException? I am also a little confused about why you throw RuntimeException in the Controller after catching SessionLoadException.

Reply:
Oh, if what we're doing is just creating a new empty session that is just creating an object, so there's no exception to handle there. For the case of loading it from a file, I handle the IO and ClassNotFound exceptions here because those are the ones thrown by the gateway, and then I turn them into the SessionLoadException because that is the one specific to this use case and that the View can handle

Reply:
Okay, makes sense thanks!